### PR TITLE
Add --print-shape debug flag for logits shape debugging

### PR DESCRIPTION
**Pull request by computerologykj-maker**
This PR introduces a small, optional debugging feature to improve observability during autoregressive text generation in OLMo-core.

## 🔧 What this PR adds

- A new `--print-shape` CLI flag in `olmo_core.generate.chat`.
- A new `print_shape: bool` option in `GenerationConfig`.
- A debug print inside `generation_module.py` that logs the shape of the logits tensor immediately after each forward pass.

This produces output such as:

[logits-shape] torch.Size([1, 1, 32000])|

when enabled.

## 🧪 Why this is useful

Understanding the exact shape of logits during generation is helpful when:

- experimenting with custom sampling strategies  
- validating batch/sequence/vocab dimensions  
- debugging shape mismatches  
- integrating external modules on top of the generation pipeline  

This feature is especially useful during development of plugins, routing layers, or generation-time tools.

## 🛡️ Safety & Backwards Compatibility

- The flag is **off by default**.
- There is **zero impact** on training, inference, or model outputs when not enabled.
- No existing behavior is modified.
- The debug print is isolated and wrapped behind a config flag.

## 📁 Files Modified

- `src/olmo_core/generate/generation_module/transformer/generation_module.py`
- `src/olmo_core/generate/generation_module/config.py`
- `src/olmo_core/generate/chat.py`

## ✔️ Summary

This PR adds a lightweight and useful debugging tool with minimal surface area and full backward compatibility.

---

> [!NOTE]
> Adds an optional `--print-shape`/`print_shape` debug path to print logits tensor shape during decoding.
> 
> - **Generation/CLI**:
>   - Adds `--print-shape` flag in `src/olmo_core/generate/chat.py` and wires it into `GenerationConfig`.
> - **Config**:
>   - Introduces `print_shape: bool` in `src/olmo_core/generate/generation_module/config.py`.
> - **Runtime (Transformer generation)**:
>   - Conditionally prints logits tensor shape after each forward pass in `src/olmo_core/generate/generation_module/transformer/generation_module.py` when `print_shape` is enabled.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ec72989e142811f4c396ce3441fb8b53c81c8334. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/olmo_core/generate/chat.py [hunk 1]
```diff
@@ -323,6 +329,7 @@ def main():
         top_p=args.top_p,
         use_cache=args.use_cache,
     )
+    generation_config.print_shape = args.print_shape
```
Comment (epwalsh) [inline src/olmo_core/generate/chat.py:332]:
```suggestion
        use_cache=args.use_cache,
        print_shape=args.print_shape,
    )
```

File: src/olmo_core/generate/generation_module/config.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ class GenerationConfig(Config):
     use_cache: bool = True
     """Whether to use an inference cache (e.g. a kv-cache) for generation."""
 
+    print_shape: bool = False
```
Comment (epwalsh) [inline src/olmo_core/generate/generation_module/config.py:41]: Could you add a docstring here?

File: src/olmo_core/generate/generation_module/transformer/generation_module.py [hunk 3]
```diff
@@ -246,6 +246,10 @@ def generate_batch(
                 cache_leftpad=cache_leftpad if generation_config.use_cache else None,
             )
 
+            # Optional debug: print logits shape
+            if getattr(generation_config, "print_shape", False):
```
Comment (epwalsh) [inline src/olmo_core/generate/generation_module/transformer/generation_module.py:250]: Using `getattr()` seems unnecessary
```suggestion
            if generation_config.print_shape:
```

**Review by epwalsh — COMMENTED** [review]
Thanks @computerologykj-maker, left a few minor suggestions. Always pinging @tyler-romero for a review.

**Comment by tyler-romero** [comment]
Thanks for your PR! But tbh I dont think this makes sense to commit to `main`. There are many values within `generate_batch` that might make sense to print for debugging info. Rather than add a flag for one of them, I think it makes sense for users to add an ad-hoc print line for the value they are interested in.
